testing/test_collect_imports.py

```diff
@@ -301,12 +307,24 @@ def _test_hook_behaviour(self) -> None:
         # ?                                  ^
         # ]
 
-        # assert len(default['reports']) == len(collect_off['reports'])
-        # for i in range(len(default['reports'])):
-        #     print("def",default['reports'][i].__dict__)
-        #     print("off",collect_off['reports'][i].__dict__)
+        for x in default["reports"]:
+            print("def", x.__dict__)
+
+        for x in collect_off["reports"]:
+            print("off", x.__dict__)
+
+        # The two above loops prints:
+
+        # def {'nodeid': '', 'outcome': 'passed', 'longrepr': None, 'result': [<Dir test_hook_behaviour0>], 'sections': []}  # noqa: E501
+        # def {'nodeid': 'tests/foo_test.py::TestDomain', 'outcome': 'passed', 'longrepr': None, 'result': [<Function test_important>], 'sections': []} # noqa: E501
+        # def {'nodeid': 'tests/foo_test.py', 'outcome': 'passed', 'longrepr': None, 'result': [<Class TestDomain>], 'sections': []} # noqa: E501
+        # def {'nodeid': 'tests', 'outcome': 'passed', 'longrepr': None, 'result': [<Module foo_test.py>], 'sections': []} # noqa: E501
+        # def {'nodeid': '.', 'outcome': 'passed', 'longrepr': None, 'result': [<Dir tests>], 'sections': []} # noqa: E501
+        # off {'nodeid': '', 'outcome': 'passed', 'longrepr': None, 'result': [<Dir test_hook_behaviour1>], 'sections': []} # noqa: E501
+        # off {'nodeid': 'src', 'outcome': 'passed', 'longrepr': None, 'result': [], 'sections': []}
+        # off {'nodeid': 'tests/foo_test.py::TestDomain', 'outcome': 'passed', 'longrepr': None, 'result': [<Function test_important>], 'sections': []} # noqa: E501
+        # off {'nodeid': 'tests/foo_test.py', 'outcome': 'passed', 'longrepr': None, 'result': [<Class TestDomain>], 'sections': []} # noqa: E501
+        # off {'nodeid': 'tests', 'outcome': 'passed', 'longrepr': None, 'result': [<Module foo_test.py>], 'sections': []} # noqa: E501
+        # off {'nodeid': '.', 'outcome': 'passed', 'longrepr': None, 'result': [<Dir src>, <Dir tests>], 'sections': []} # noqa: E501
```

Comment:
At this point, I need some guidance.

The problem now is that while the CollectReport's for classes and functions are correct, the directory gets a report (even though no classes/functions were collected from it). What should be the expected behaviour in this case? Since the directory will be collected before any of the classes/functions I would need to filter it out later on somehow. Not sure what the correct approach is here. 

I do feel like collecting the report for the directory (even though it's not "used") makes sense but I would need a maintainer to check.

Reply:
> The problem now is that while the CollectReport's for classes and functions are correct, the directory gets a report (even though no classes/functions were collected from it). What should be the expected behaviour in this case?

This is fine -- currently on `main` if we implement `pytest_collectreport` and run pytest on a bunch of directories without any tests, we get collection reports for all directories, even though no teste items were collected. 

My previous concern was that we were generating collection reports for items that were in fact being discarded later, and that would definitely be  a problem -- but collection reports for directories it is expected that some of them might turn out to not collect anything.